src/Main.elm

```diff
@@ -1,10 +1,571 @@
+
+        GoUp ->
+            let
+                newZipperStep : Maybe (Zipper.Zipper Node)
+                newZipperStep =
+                    model.zipperStep
+                        |> Maybe.andThen Zipper.parent
+
+                updatedTree : Maybe (Tree Node)
+                updatedTree =
+                    newZipperStep
+                        |> Maybe.map Zipper.tree
+            in
+            ( { model | zipperStep = newZipperStep, currentTree = updatedTree }, Cmd.none )
+
+
+
+-- VIEW
+
+
+type alias Model =
+    { directory : Tree Node
+    , currentTree : Maybe (Tree Node)
+    , zipperStep : Maybe (Zipper.Zipper Node)
+    }
+
+
+view : Model -> Html Msg
+view model =
+    let
+        nodes : List Node
+        nodes =
+            case model.currentTree of
+                Just tree ->
+                    getChildrenNodes tree
+
+                Nothing ->
+                    []
+    in
+    div [ class "w-screen h-screen bg-gray-50" ]
+        [ div [ class "w-full h-auto" ]
+            [ viewBackButton model
+            , div []
+                (List.append
+                    (viewNodes nodes ViewFolder)
+                    (viewNodes nodes ViewFile)
+                )
+            ]
+        ]
+
+
+viewBackButton : Model -> Html Msg
+viewBackButton model =
+    let
+        hasParent : String
+        hasParent =
+            case parentLabel of
+                Just (Dir id audienceFolder) ->
+                    audienceFolder.name
+
+                Just Root ->
+                    "Home"
+
+                _ ->
+                    ""
+
+        parentLabel : Maybe Node
+        parentLabel =
+            model.zipperStep
+                |> Maybe.andThen Zipper.parent
+                |> Maybe.map Zipper.label
+    in
+    case String.length hasParent > 0 of
+        True ->
+            div [ class "w-full bg-gray-800 px-4 py-4 cursor-pointer", onClick GoUp ]
+                [ span [ class "text-white " ] [ FA.icon FA.arrowAltLeft ]
+                , span [ class "inline-block ml-2 text-white font-bold" ] [ text hasParent ]
+                ]
+
+        False ->
+            text ""
+
+
+type ViewType
+    = ViewFolder
+    | ViewFile
+
+
+viewNodes : List Node -> ViewType -> List (Html Msg)
+viewNodes nodes viewType =
+    let
+        nodesToBeViewed : List Node
+        nodesToBeViewed =
+            case viewType of
+                ViewFolder ->
+                    List.filter foldersOnly nodes
+
+                ViewFile ->
+                    List.filter filesOnly nodes
+
+        name : Node -> String
+        name node =
+            case node of
+                Dir id audienceFolder ->
+                    audienceFolder.name
+
+                File id audience ->
+                    audience.name
+
+                Root ->
+                    ""
+    in
+    nodesToBeViewed
+        |> List.map (\node -> viewNode (name node) viewType (GoTo node))
```

Comment:
The `viewNode` function could run `name` itself, it doesn't need to be drip-fed that name pre-prepared :)